src/instantsearch/instantsearch.ts

```diff
@@ -28,6 +29,7 @@ export type InstantSearchConfig = {
     appId: string,
     apiKey: string
   ) => object;
+  searchClient?: object;
```

Comment:
With TypeScript we can create new type to better represent our program. Instead of rely on the `object` type we can create a type that better represent what we expect.

Reply:
It wasn't done for `searchParameters` though, I was wondering if there was a reason, @iam4x?

Otherwise, I will add a more strict type for `searchClient`.

Reply:
Indeed, for the `SearchParameters` it's a bit more tedious because the data structure is way more larger. But we should add the correct types at some point.

Reply:
`searchParameters` are what the Algolia API expect, it's hard to type since it can change at any time without our control.

Here you should create a type and use it :+1:

Reply:
Makes sense. I added a type. Let me know if it can be even more strict. I'm expecting this format: `Promise.resolve({ results: [] })`

Reply:
You can import searchParameters from the typing of `algoliasearch`